packages/sdk-v2/src/sdk/goodid/sdk.ts

```diff
@@ -0,0 +1,65 @@
+import { Envs } from "../constants";
+import { CertificateItem } from "./types";
+
+const getBearerToken = async (baseEnv: string, fvsig: string, account: string, type: "location" | "identity") => {
+  const devEnv = baseEnv === "fuse" ? "development" : baseEnv;
+  const { backend } = Envs[devEnv];
+
+  const endpoint = `${backend}/goodid/certificate/${type}`;
+  const authEndpoint = `${backend}/auth/fv2`;
+  const { token } = await fetch(authEndpoint, {
+    method: "POST",
+    headers: { "content-type": "application/json" },
+    body: JSON.stringify({ fvsig, account })
+  }).then(_ => _.json());
+
+  return { token, endpoint };
+};
+
+export const requestLocationCertificate = async (
+  baseEnv: string,
+  [lat, long]: [number, number],
+  fvSig: string,
+  account: string
+): Promise<CertificateItem> => {
+  const { endpoint, token } = await getBearerToken(baseEnv, fvSig, account, "location");
+
+  return fetch(endpoint, {
```

Comment:
1. maybe create some util wrapping calls to our backend. something like I've done at etoro app: https://github.com/GoodDollar/GoodWalletV2/blob/master/packages/utils/index.ts
2. first  you do res.json
3. then you res res.ok. if res.ok = false then you throw new Error(res.message ?? res.error)

Reply:
at least you could define something like 

```javascript
const asG$Respose = fetchResponse => fetchResponse.json().then(({ success, ok, error, message, ...response }) => {
  if (!success && !ok) {
    throw new Error(error ?? message ?? "Unknown GoodServer error")
  }

  return response
})

// usage
fetch(...).then(asG$Response)
```

Reply:
```javascript

const g$Request = (json, method = 'POST') => ({
  method,
  headers: { "content-type": "application/json" },
  body: JSON.stringify(json)
})

// usage
const { token } = await fetch(authEndpoint, g$Request({ fsig, account }).then(g$Response)
```

Reply:
implemented here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/136/commits/f79359a2cc60f3409328fd0fb1370bd7078ec57e